app/Services/Model/Imp/ScheduleService.php

```diff
@@ -0,0 +1,262 @@
+use models\summit\PresentationCategory;
+use models\summit\Summit;
+use models\summit\SummitEvent;
+use models\summit\SummitProposedSchedule;
+use models\summit\SummitProposedScheduleSummitEvent;
+
+/**
+ * Class ScheduleService
+ * @package App\Services\Model
+ */
+final class ScheduleService
+extends AbstractPublishService implements IScheduleService
+{
+    /**
+     * @var ISummitProposedScheduleRepository
+     */
+    private $schedule_repository;
+
+    /**
+     * @var ISummitEventRepository
+     */
+    private $event_repository;
+
+    /**
+     * ScheduleService constructor.
+     * @param ISummitProposedScheduleRepository $schedule_repository
+     * @param ISummitEventRepository $event_repository
+     * @param ITransactionService $tx_service
+     */
+    public function __construct(ISummitProposedScheduleRepository $schedule_repository,
+                                ISummitEventRepository $event_repository,
+                                ITransactionService $tx_service)
+    {
+        parent::__construct($schedule_repository, $tx_service);
+        $this->schedule_repository = $schedule_repository;
+        $this->event_repository = $event_repository;
+    }
+
+    private function isAuthorizedUser(Member $member, Summit $summit, PresentationCategory $category): bool {
+        if ($member->isAdmin()) return true;
+        if ($member->isSummitAdmin()) return true;
+        return $summit->isTrackChair($member, $category);
+    }
+
+    /**
+     *@inheritDoc
+     */
+    public function publishProposedActivityToSource(
+        string $source, int $presentation_id, array $payload):SummitProposedScheduleSummitEvent {
+
+        $schedule = $this->tx_service->transaction(function () use ($source, $presentation_id, $payload) {
+
+            $event = $this->event_repository->getById($presentation_id);
+            if (!$event instanceof Presentation)
+                throw new EntityNotFoundException("event id {$presentation_id} does not exists!");
+
+            $schedule = null;
+            $schedules = $this->schedule_repository->getBySourceAndSummitId($source, $event->getSummitId());
+
+            $selection_plan = $event->getSelectionPlan();
+            if (!$selection_plan instanceof SelectionPlan)
+                throw new EntityNotFoundException("event id {$presentation_id} does not have a selection plan");
+
+            if (!$selection_plan->isAllowProposedSchedules())
+                throw new ValidationException("selection plan id {$selection_plan->getId()} does not allow proposed schedules");
+
+            if (count($schedules) == 0) {
+                $member = ResourceServerContext::getCurrentUser(false);
+                $schedule = new SummitProposedSchedule();
+                $schedule->setSource($source);
+                $schedule->setSummit($event->getSummit());
+                if (isset($data['schedule_name'])) {
+                    $schedule->setName($data['schedule_name']);
```

Comment:
data does no exits on this context lets remove it in favor or the set name bellow

Reply:
fixed